test_runner/regress/test_tenants.py

```diff
@@ -122,3 +124,46 @@ def test_metrics_normal_work(neon_env_builder: NeonEnvBuilder):
         log.info(
             f"process_start_time_seconds (UTC): {datetime.fromtimestamp(metrics.query_one('process_start_time_seconds').value)}"
         )
+
+
+def test_pageserver_metrics_removed_after_detach(neon_env_builder: NeonEnvBuilder):
+    """Tests that when a tenant is detached, the tenant specific metrics are not left behind"""
+
+    neon_env_builder.num_safekeepers = 3
+
+    env = neon_env_builder.init_start()
+    tenant_1, _ = env.neon_cli.create_tenant()
+    tenant_2, _ = env.neon_cli.create_tenant()
+
+    env.neon_cli.create_timeline("test_metrics_removed_after_detach", tenant_id=tenant_1)
+    env.neon_cli.create_timeline("test_metrics_removed_after_detach", tenant_id=tenant_2)
+
+    pg_tenant1 = env.postgres.create_start("test_metrics_removed_after_detach", tenant_id=tenant_1)
+    pg_tenant2 = env.postgres.create_start("test_metrics_removed_after_detach", tenant_id=tenant_2)
+
+    for pg in [pg_tenant1, pg_tenant2]:
+        with closing(pg.connect()) as conn:
+            with conn.cursor() as cur:
+                cur.execute("CREATE TABLE t(key int primary key, value text)")
+                cur.execute("INSERT INTO t SELECT generate_series(1,100000), 'payload'")
+                cur.execute("SELECT sum(key) FROM t")
+                assert cur.fetchone() == (5000050000,)
+
+    def get_ps_metric_samples_for_tenant(tenant_id: ZTenantId) -> List[Sample]:
+        ps_metrics = parse_metrics(env.pageserver.http_client().get_metrics(), "pageserver")
+        samples = []
+        for metric_name in ps_metrics.metrics:
+            for sample in ps_metrics.query_all(
+                name=metric_name, filter=dict(tenant_id=str(tenant_id))
```

Comment:
nit: Lets use dict literal instead: 

```python
filter={"tenant_id": str(tenant_id}
```
This is the common style across the codebase. And by the least importance it is one bytecode instruction smaller.  dict(...) is two instructions: lookup `dict` in globals and a function call, and literal is just a single BUILD_MAP instruction:

```
In [1]: import dis

In [2]: dis.dis("foo = {}")
  1           0 BUILD_MAP                0
              2 STORE_NAME               0 (foo)
              4 LOAD_CONST               0 (None)
              6 RETURN_VALUE

In [3]: dis.dis("foo = dict()")
  1           0 LOAD_NAME                0 (dict)
              2 CALL_FUNCTION            0
              4 STORE_NAME               1 (foo)
              6 LOAD_CONST               0 (None)
              8 RETURN_VALUE

```